frontend/src/lib/components/Badges/Badge.svelte

```diff
@@ -0,0 +1,23 @@
+﻿<script lang="ts">
+	import { Badge } from '.';
+
+	export let type = '';
+	export let button = false;
+	export let icon = '';
+</script>
+
+{#if button}
+	<button on:click class="hover:drop-shadow-xl hover:brightness-90">
+		<Badge {type} {icon}>
```

Comment:
I'm not really sure I like this. Can we come up with another option other than a recursive component (also a recursive component should use `<svelte:self>`)? Same with the TaggedBadge.
Maybe we just have 2 components. A Badge and a ButtonBadge, the button obviously wrapping the Badge. 

I could be sold on this though, I just want to make sure we think through this first.

Reply:
I know it's a bit unique, but...I think it's somewhat elegant. 🤷
`TaggedButtonBadge` 🤔. I don't care too much either way.